Add docs related to PE view (#1708)
wso2/docs-choreo-dev: en/pe-docs/docs/integrate-and-manage-external-services/integrate-and-manage-gen-ai-services.md

# Integrate and Manage Generative AI Services

Generative AI (GenAI) services leverage advanced machine learning models to create original content such as text, images, music, or code, by identifying and learning patterns from existing data. Powered by deep neural networks and other machine learning models, these services can generate human-like outputs in various formats. This versatility makes GenAI ideal for tasks such as content creation, image generation, conversation automation, etc. 

> Choreo enables seamless integration with GenAI services, allowing you to incorporate these capabilities within your applications.

Reviewer: ```suggestion
Choreo enables seamless integration with GenAI services, allowing the developers to incorporate these capabilities within their applications.
```

## Register a GenAI service

> To use a GenAI service in Choreo, you need to register it. Once registered, the service becomes available in the Internal Marketplace, allowing you to consume it via a Connection.

Reviewer: ```suggestion
To use a GenAI service in your Choreo organization/project, you need to register it. Once registered, the service becomes available in the Internal Marketplace, allowing the developers to consume it.
```

Author: ```
To seamlessly integrate a GenAI service into an application deployed in Choreo, you need to register it. Once registered, the service becomes available in the Internal Marketplace, allowing developers to consume it via a Connection.

```

!!! note 
     Ensure that you register the service at the appropriate level depending on the usage. If the service must be shared among multiple projects, register it at the organization level. Otherwise, register it at the project level.

### Step 1: Select a service provider

1. Sign in to the [Choreo Console](https://console.choreo.dev/).
2. Follow one of these steps depending on your requirements:
    - To register a GenAI service at the organization level, go to the Choreo Console header and select your organization from the **Organization** list. 
    - To register a GenAI service at the project level, go to the Choreo Console header and select your project from the **Project** list. 
3. In the left navigation menu, click **DB & Services** and then click **GenAI Services**.
4. Click **+ Register**.
5. Select a service provider.
6. Click **Next**.

### Step 2: Provide service details

1. Under **Register Service**, enter the following details for the service:
    - A **Name** and **Version**.
    - The **Service URL**.

2. Click **Next**.

### Step 3: Add configurations

1. Under **Add Configurations**, enter details for the service.

>     !!! note 
>          - The configuration details to enter vary depending on the selected service provider.
>          - To successfully register a service, ensure you provide all the required configurations.

Reviewer: ```suggestion

!!! note 
        - The configuration details to enter vary depending on the selected service provider.
        - To successfully register a service, ensure you provide all the required configurations.
```

2. Click **Register**.

Once registered, the GenAI service is automatically listed in the Internal Marketplace.

If you want to remove the service from the Internal Marketplace, see [Remove a GenAI service from the Internal Marketplace](#remove-a-genai-service-from-the-internal-marketplace).

> ## Discover GenAI services

Reviewer: Let's remove discover part.

The GenAI services you register are discoverable via the Internal Marketplace for you to consume via a Connection.

For details on consuming a GenAI service via a connection, see [Create a Connection](../develop-components/sharing-and-reusing/create-a-connection.md).

## Manage GenAI services

When you create a GenAI service, it gets listed in the **GenAI Services** list. 

### View or update GenAI service details

To view or update a GenAI service, follow these steps.

1. Sign in to the [Choreo Console](https://console.choreo.dev/).
2. In the left navigation menu, click **DB & Services** and then click **GenAI Services**. This lists all the GenAI services you have created.
3. Click on a service to view or update its details.
    - **General Details**: Displays service metadata. For example, the service name, overview, labels, etc. 
    - **Service Definition**: Displays the service definition. To update the definition, click **Upload** and select the new definition file.

> ### Add a GenAI service to the Internal Marketplace

Reviewer: I guess this is also not relevant now. (Once you register, it seems it is already added to marketplace)

Author: It gets automatically added, but it can be added again if removed.
